src/Client/src/Asp.Versioning.Http.Client/System.Net.Http/HttpResponseMessageExtensions.cs

```diff
@@ -69,6 +70,67 @@ public static SunsetPolicy ReadSunsetPolicy( this HttpResponseMessage response )
         return policy;
     }
 
+    /// <summary>
+    /// Gets an API deprecation policy from the HTTP response.
+    /// </summary>
+    /// <param name="response">The <see cref="HttpResponseMessage">HTTP response</see> to read from.</param>
+    /// <returns>A new <see cref="DeprecationPolicy">deprecation policy</see>.</returns>
+    public static DeprecationPolicy ReadDeprecationPolicy( this HttpResponseMessage response )
+    {
+        ArgumentNullException.ThrowIfNull( response );
+
+        var headers = response.Headers;
+        var date = default( DateTimeOffset );
+        DeprecationPolicy policy;
+
+        if ( headers.TryGetValues( Deprecation, out var values ) )
+        {
+            var culture = CultureInfo.CurrentCulture;
+
+            foreach ( var value in values )
+            {
+                var split = value.Trim( '@' );
+                if ( long.TryParse( split, out var unixTimestamp ) )
```

Comment:
Unfortunately, I think this is another place where a compiler directive is required because the newer versions of .NET can use a span here.

```c#
#if NETSTANDARD
if ( long.TryParse( value.Substring( 1 ), out var unixTimestamp ) )
#else
if ( long.TryParse( value.AsSpan()[1..], out var unixTimestamp ) )
#endif
```